### PR TITLE
TST: Add Python dev job to cron job

### LINKED ISSUE #9201: TST: Add cron to test against Python dev
I wonder if there is value in adding a cron job to test against Python dev. Stemmed from #9198 . cc @astrofrog and @bsipocz

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to add a test job for dev version of Python.

Fixes #9201

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -95,9 +95,9 @@ matrix:
 
         # Now try with all optional dependencies.
         - language: python
-          python: 3.8
+          python: nightly
           dist: bionic
-          name: Python 3.8 with all optional dependencies
+          name: Python dev with all optional dependencies
```
Comment (bsipocz) [inline .travis.yml:100]: I would go with mandatory dependencies only. We should be able to test even when some of our upstream have issues

Comment (bsipocz) [reply .travis.yml:100]: also, you need to update the tox stuff below

**Comment by pllim** [comment]
I don't know how to read this tox thingy. How do I define Python dev in tox language? It is complaining about `python3.8` yadda yadda.

**Comment by pllim** [comment]
This should work... See https://travis-ci.org/github/astropy/astropy/jobs/670756127 for example log when I ran it for the PR. 

```
platform linux -- Python 3.9.0a5+, pytest-5.3.5, py-1.8.1, pluggy-0.13.1
```

**Comment by bsipocz** [comment]
Looks perfect. The only issue is with this line, but it's upstream issue.

I'm merging without waiting for the jobs to finish the cron test won't run on CI anyway.

**Comment by bsipocz** [comment]
Something is not right here: https://travis-ci.org/github/astropy/astropy/jobs/670756127#L373

Screen Shot 2020-04-03 at 14 57 57

**Comment by pllim** [comment]
Yay, it worked!

https://travis-ci.org/github/astropy/astropy/jobs/670967438

```
platform linux -- Python 3.9.0a5+, pytest-5.3.5, py-1.8.1, pluggy-0.13.1
cachedir: .tox/pydev-test/.pytest_cache

Running tests with Astropy version 4.1.dev909+g7e6e4b90f.
Running tests in astropy docs.
...
========= 13023 passed, 1140 skipped, 93 xfailed in 174.99s (0:02:54) ==========
```

If this still let things slip through, might need to add optional dependencies and/or remote data in the future. 👀

**Pull request merged**